lib/src/facts/aix/memory_resolver.cc

```diff
@@ -0,0 +1,47 @@
+#include <internal/facts/aix/memory_resolver.hpp>
+#include <internal/util/aix/odm.hpp>
+
+#include <sys/vminfo.h>
+#include <sys/cfgodm.h>
+#include <system_error>
+
+using namespace std;
+using namespace facter::util::aix;
+
+namespace facter { namespace facts { namespace aix {
+    memory_resolver::data memory_resolver::collect_data(collection& facts)
+    {
+        data result;
+
+        vminfo info;
+        auto res = vmgetinfo(&info, VMINFO, sizeof(info));
+        if (res < 0) {
+            throw system_error(errno, system_category());
+        }
+        result.mem_total = info.memsizepgs * 4096;
```

Comment:
I should add that IBM's documentation points to the comments in the headers as being authoritative, so I trust it's not going to change anytime soon